src/main/java/org/opensearch/geospatial/index/common/shape/ShapeConverter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.common.shape;
+
+import static org.apache.commons.lang3.ArrayUtils.toPrimitive;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+
+import org.apache.lucene.geo.XYLine;
+import org.apache.lucene.geo.XYPolygon;
+import org.opensearch.geometry.Line;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geometry.ShapeType;
+
+// Utility class to convert compatible shapes from opensearch to Lucene
+public class ShapeConverter {
+
+    public static XYLine toXYLine(Line line) {
+        Objects.requireNonNull(line, String.format(Locale.getDefault(), "%s cannot be null", ShapeType.LINESTRING));
+        float[] x = toFloatArray(line.getX());
+        float[] y = toFloatArray(line.getY());
+        return new XYLine(x, y);
+    }
+
+    public static XYPolygon toXYPolygon(Rectangle rectangle) {
+        Objects.requireNonNull(rectangle, "Rectangle cannot be null");
+        // build polygon by assigning points in Counter Clock Wise direction (default for polygon) and end at where
+        // you started since it has to be linear ring
+        // (minX,minY) -> (maxX, minY) -> (maxX, maxY) -> (minX, maxY) -> (minX, minY)
+        double[] x = new double[] {
+            rectangle.getMinX(),
+            rectangle.getMaxX(),
+            rectangle.getMaxX(),
+            rectangle.getMinX(),
+            rectangle.getMinX() };
+        double[] y = new double[] {
+            rectangle.getMinY(),
+            rectangle.getMinY(),
+            rectangle.getMaxY(),
+            rectangle.getMaxY(),
+            rectangle.getMinY() };
+        return new XYPolygon(toFloatArray(x), toFloatArray(y));
+    }
+
+    public static XYPolygon toXYPolygon(Polygon polygon) {
+        Objects.requireNonNull(polygon, String.format(Locale.getDefault(), "%s cannot be null", ShapeType.POLYGON));
+        XYPolygon[] holes = buildXYPolygonFromHoles(polygon);
+        Line line = polygon.getPolygon();
+        XYLine polygonLine = toXYLine(line);
+        return new XYPolygon(polygonLine.getX(), polygonLine.getY(), holes);
+    }
+
+    private static XYPolygon[] buildXYPolygonFromHoles(final Polygon polygon) {
+        return IntStream.range(0, polygon.getNumberOfHoles())
+            .mapToObj(i -> polygon.getHole(i))
```

Comment:
use lamda? `polygon::getHole`

Reply:
Good catch. Changed to method references from lambda.